charts/gha-runner-scale-set/templates/_helpers.tpl

```diff
@@ -96,14 +96,20 @@ volumeMounts:
 {{- end }}
 
 {{- define "gha-runner-scale-set.dind-container" -}}
-image: docker:dind
+{{- range $i, $container := .Values.template.spec.containers }}
+  {{- if eq $container.name "dind" }}
```

Comment:
How does this default the values if the "dind" container name is not found?

Reply:
Nice catch, update please re-review